[CS2113-T11-4] NUS To-do list

clement559 opened the pull request:
With the transition to Canvas, the most important feature of LumiNUS’s deadline reminders is gone! NUS To-do list aims to bring an application to keep students  aware of their deadlines and not miss them.

choongzhanhong commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,298 @@
+
+### Ui component
+
+The code for this component is found in [`Ui.java`](../src/main/java/seedu/todolist/ui/Ui.java).
```
I think this link 404's? But great that you're using relative path.
Same for the following links below.

choongzhanhong commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,298 @@
+
+### Architecture
+
+![ArchitectureDiagram](images/ArchitectureDiagram.png)
```
I really like the extra formatting and style put into the diagrams. They're really thorough too.

choongzhanhong replied:
Overall, I like the extra thought on the styling and user-friendliness of the diagrams. The DG is also very well written. Just to note, I'm still not too familiar with PUML but if it's possible to share the CSS/Styles across diagrams, would that be good for consistency?

choongzhanhong commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,298 @@
+Step 2. The user executes `list` command. The `list` command calls `Ui#printTaskList()` which lists all tasks in the 
+task list. `ToDoListManager` calls `storage#saveData()`, so the task list is saved into `'./data.txt'`.
+
+![Step2](images/StorageSequenceDiagramStep2-Step_2.png)
```
I think this diagram could be broken up into two pictures and/or place more focus on the methods more closely related to `list` command, as after a few diagrams, it may be redundant to showcase the user and parser.

choongzhanhong replied:
Also, I am slightly confused at Step 7 onwards in this diagram. Should there be return values by `TaskList`?

EangJS commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,298 @@
+
+### Architecture
+
+![ArchitectureDiagram](images/ArchitectureDiagram.png)
```
Architecture Diagram is very minimalist and clear which makes it very easy to understand. 👍

EangJS commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,298 @@
+`Command` object, which will act on the `TaskList` if needed and output the result of the command through the `Ui`
+component.
+
+### TaskList component
+
+The code for this component is found in [`TaskList.java`](../src/main/java/seedu/todolist/task/TaskList.java).
+
+### Storage component
+
+The code for this component is found in [`Storage.java`](../src/main/java/seedu/todolist/storage/Storage.java).
+
+The Storage component can save the task list as TaskList objects in a .txt file format using Serialization and read it 
+back into a TaskList object.
+
```
Maybe can add some class diagrams for the these components?

EangJS commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,298 @@
+
+The following sequence diagram shows how the delete task operation works:
+
+![DeleteTaskCommandSequence](images/DeleteTaskCommandSequence.png)
```
I like this Sequence diagram, The information is very clear and the colors make it very understand too!

EangJS commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,298 @@
+
+The following sequence diagram shows how the edit operations works:
+
+![EditDeadlineCommandSequence](images/EditDeadlineCommandSequence.png)
```
This sequence diagram seems to be missing activity blocks. 😲

EangJS commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,298 @@
+Step 2. The user executes `list` command. The `list` command calls `Ui#printTaskList()` which lists all tasks in the 
+task list. `ToDoListManager` calls `storage#saveData()`, so the task list is saved into `'./data.txt'`.
+
+![Step2](images/StorageSequenceDiagramStep2-Step_2.png)
```
The steps 1,2,3,4,... seems redundant, the sequence should be clear enough.
Maybe can also standardise the coloring?

EangJS commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,298 @@
+#### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+With the transition to Canvas, the most important feature of LumiNUS’s deadline reminders is gone! Our project aims to 
+bring an application to keep you aware of your deadlines and not miss them.
```
Very cool concept, I like this program idea!

EangJS commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,298 @@
+
+### [Proposed] History feature
+
+The proposed history feature is facilitated by the `Storage`, `TaskList` and `Command` classes. Internally, there will 
+be 2 task lists stored - `completedTasks` and `uncompletedTasks`. There will be a rework to how marking tasks as done 
+works, a removal of the operation `TaskList#setDone()` and a new command for users to input to the CLI: `history`.
+
+There will be 2 new operations:
+* `TaskList#markTask(id i)` - Moves the task at id i of `uncompletedTasks` to `completedTasks`.  
+* `TaskList#unmarkTask(id i)` - Moves the task at id i of `completedTasks` to `uncompletedTasks`.
+
+Given below is an example usage scenario and how the history mechanism works.
```
Seems like a very intuitive feature to have! Can't wait to see it! 👍

EangJS reviewed (commented):
# Overall a very well done DG.  💯 
* Maybe can add some class diagrams to the major components.
* Touch up on the sequence diagrams and you should be good to go! 😄

jerome2406 committed:
Delete unused files

jerome2406 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp

clement559 committed:
Added edit deadline and (upcoming) completion history feature
Fixed minor errors

jerome2406 committed:
Update user guide for progress bar feature and correct some typos

jerome2406 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp into branch-developerGuide3

# Conflicts:
#	docs/diagrams/StorageSequenceDiagramStep1.puml

jerome2406 committed:
Fix spacing in developer guide

clement559 committed:
Merge pull request #69 from jeromeongithub/branch-userGuide

Branch user guide

jerome2406 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp into progressBarImprovements

# Conflicts:
#	docs/diagrams/StorageSequenceDiagramStep1.puml

clement559 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp into branch-UG_DG

# Conflicts:
#	docs/UserGuide.md

clement559 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp into branch-UG_DG

KedrianLoh committed:
updated user guide

jeromeongithub committed:
Merge pull request #68 from clement559/branch-UG_DG

UG Addition

KedrianLoh committed:
Merge branch 'master' into update_UG

KedrianLoh committed:
Merge pull request #70 from KedrianLoh/update_UG

updated user guide

erjunze committed:
Merge pull request #66 from jeromeongithub/branch-developerGuide3

Branch developer guide3

erjunze committed:
Merge pull request #67 from jeromeongithub/progressBarImprovements

Progress bar improvements

KedrianLoh committed:
amend user guide

KedrianLoh committed:
Merge pull request #71 from KedrianLoh/amend_UG

amend user guide

erjunze committed:
Fix minor bug in user guide

erjunze committed:
Merge pull request #72 from ERJUNZE/UserGuide

Fix minor bug in user guide

erjunze committed:
Fix missing newline causing formatting issues

ChongQiRong commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
+
+### Logic component
+
+The main code for this component is found in [`Parser.java`](../src/main/java/seedu/todolist/logic/Parser.java).
```
Any reason why only the logic component uses the phrase "main code" while the uses "code"? Perhaps could make it consistent by standardizing to "code"?

azriellee commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
 # Developer Guide
 
+## Table of Contents
+
+* Table of Contents
+{:toc}
```
Nice use of a Table of Contents, allows me to understand the entire flow and structure well. However, for the 'Product Scope' might be more appropriate to include those details at the start? Allows me to understand who the program targets and what the program can do, giving me more context to understand subsequent diagrams and logic

linuspuah replied:
Great overall formatting and table of contents

azriellee commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
+1. **Configure the JDK**: Follow the guide [_[se-edu/guides] IDEA: Configuring the JDK_](https://se-education.org/guides/tutorials/intellijJdk.html) to ensure Intellij is configured to use **JDK 11**.
+1. **Import the project as a Gradle project**: Follow the guide [_[se-edu/guides] IDEA: Importing a Gradle project_](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into IDEA.<br>
+   - Note: Importing a Gradle project is slightly different from importing a normal Java project.
+1. **Verify the setup**: Run `Main` and try a few commands.
```
Could maybe give more guidance on the setting up? More in-depth instructions and maybe including some screenshots will be great!

azriellee commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
+
+### Architecture
+
+![ArchitectureDiagram](images/ArchitectureDiagram.png)
```
I agree with the reviewers above, the architecture diagram is neatly done and very understandable! Nice!

ChongQiRong commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
+
+The following sequence diagram shows how the edit operations works:
+
+![EditDeadlineCommandSequence](images/EditDeadlineCommandSequence.png)
```
There seems to be missing activation bars. 
![image](https://user-images.githubusercontent.com/88194809/228704799-906b6851-d809-4fa0-890b-6e9b00019d59.png)
I think there a missing "(" for .FLAG()

azriellee commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
+
+The code for this component is found in [`ToDoListManager.java`](../src/main/java/seedu/todolist/ToDoListManager.java).
+
+![ManagerClassDiagram](images/ManagerClassDiagram.png)
```
Is this a class diagram? Would be good to label it as a class diagram in the DG

azriellee commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
+
+The following sequence diagram shows how the mark/unmark task operation works:
+
+![MarkOrUnmarkTaskCommandSequence](images/MarkorUnmarkTaskCommandSequence.png)
```
![image](https://user-images.githubusercontent.com/88471489/228705691-5f16e459-f1f0-464e-a022-121631806570.png)
Should Mark/UnmarkTaskCommand return arrow be dashed instead of a solid arrow?

ChongQiRong replied:
![image](https://user-images.githubusercontent.com/88194809/228707077-5c238e4e-c257-498a-a0a2-c96fc4605be7.png)
Should the return arrow be dotted line instead?

azriellee commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
+Scenario 1: The user launches the application and there is a valid save file found. Storage loads the file successfully
+and informs the user of the loaded task list (and how many tasks there are in it).
+
+![StartUpStorageSequenceWithValidSaveFile](images/StartUpStorageWithValidSaveFile.png)
```
The distance between main and user seems rather long, maybe the diagram could be resized to fit more appropriately? Will be good to include activation bar for main as well

ChongQiRong commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
 # Developer Guide
```
Overall, I like the sequence and architecture diagrams. Perhaps can consider adding in class diagrams as well

azriellee commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
+Scenario 2: The user launches the application and there is no save file found. Storage informs the user that there is no
+save file found and that a new one will be created for them later.
+
+![StartUpStorageSequenceWithNoSaveFile](images/StartUpStorageSequenceWithNoSaveFile.png)
```
Activation bars are not deactivated properly, please deactivate them accordingly!

azriellee reviewed (commented):
Overall, the DG is very clean and concise! The use of diagrams is very good to complement the information given in the DG.

erjunze committed:
Add in missing commands to user guide

erjunze committed:
Merge pull request #73 from ERJUNZE/UserGuide

Fix missing newline causing formatting issues

linuspuah commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
+
+## Design
+
+> The `.puml` files used to create the diagrams in this guide can be found in the [diagrams](diagrams) folder.
```
The link to the diagrams folder seems to give an error

linuspuah commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
+
+### Storage component
+
+The code for this component is found in [`Storage.java`](../src/main/java/seedu/todolist/storage/Storage.java).
```
The links to tasklist and storage doesnt work

linuspuah commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
+
+This ListTasksCommand extends NUS To-Do List with an automatic sorting feature that sorts all tasks in an ascending
+deadline order and displays the To-Do List to users. It is facilitated by the TaskList, Command class, and UI class.
+It implements the `TaskList#toString()` and  `ui#printTaskList()`operation.
```
from my understanding "#" is used to separate class and methods for java docs, however for developer guides I am unsure about its use.

linuspuah commented on docs/DeveloperGuide.md:
```diff
@@ -1,35 +1,306 @@
+| v2.0    | user     | set a task to repeat                                                                                             | create 1 task to represent repeating tasks every week                     |
+| v2.0    | user     | set priority level and can sort the tasks based on the priority level                                            | identify high priority tasks                                              |
+| v2.0    | user     | see a progress bar                                                                                               | track my progress of unfinished tasks                                     |
+| v2.0    | user     | view up to 10 previously completed tasks tied with the completion date and time                                  | track my progress of finished tasks                                       |
```
remember to add v2.1 user stories

erjunze committed:
Fix user guide again

erjunze committed:
Fix checkstyle issues

erjunze committed:
Merge pull request #75 from ERJUNZE/UserGuide

Fix user guide again

erjunze committed:
Fix some easy-to-fix bugs

erjunze committed:
Shorten a method, fix a bug

clement559 committed:
Merge pull request #114 from ERJUNZE/BugFixes

Bug fixes

clement559 committed:
Added repeating task feature to DG

clement559 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp

clement559 committed:
Removed checkRepeatingTaskCommand, checkRepeatingTask function now runs every time after a command is executed.
Disallowed users to input dates and times that have already passed. New exception created.
Added details in user guide

clement559 committed:
Forgot to stage the files for commit.

clement559 committed:
Remove usage for repeatingTaskCommand

clement559 committed:
Import PassedDateException

erjunze committed:
Merge pull request #115 from clement559/branch-UG_DG

Added repeating task feature to DG

clement559 committed:
Currently requires users to input a filter of c or nc after the list command

clement559 committed:
Fix minor bugs as pointed out by Jun Ze

KedrianLoh committed:
Added find error

KedrianLoh committed:
Added flags for findbypriority and findbytag

KedrianLoh committed:
Updated Parser and ParserUtil

KedrianLoh committed:
Updated TaskList

KedrianLoh committed:
Updated Ui

KedrianLoh committed:
Added InvalidFindException class

KedrianLoh committed:
Added FindByTag class

KedrianLoh committed:
Update FindByTag class

KedrianLoh committed:
Merge branch 'master' into add_find_function

KedrianLoh committed:
Fixed minor errors

KedrianLoh committed:
Revert "Fixed minor errors"

This reverts commit 8e204b6fca882591b3b0d28bfa6b50f42fccd12e.

KedrianLoh committed:
Merge branch 'master' into add_find_function

# Conflicts:
#	src/main/java/seedu/todolist/logic/ParserUtil.java
#	src/main/java/seedu/todolist/task/TaskList.java

KedrianLoh committed:
Merge branch 'add_find_function' of https://github.com/KedrianLoh/tp into add_find_function

# Conflicts:
#	src/main/java/seedu/todolist/logic/ParserUtil.java
#	src/main/java/seedu/todolist/task/TaskList.java

KedrianLoh committed:
Removed unused import

KedrianLoh committed:
Merge pull request #117 from KedrianLoh/add_find_function

Add find function

clement559 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp into branch-FilterCompletion

clement559 committed:
Added filters for listing of tasks by done, undone or overdue.
Listing of tasks may still be done without a filter for full task list.

clement559 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp into branch-BugFixing

clement559 committed:
Changed usage of string filter to predicate.

clement559 committed:
Removed test case for minus 1 week as the program does not input past dates/times

clement559 committed:
Remove unused imports, try to fix github check errors

clement559 committed:
Changed test cases again

clement559 committed:
Edited formatting and indentation

erjunze committed:
Merge pull request #116 from clement559/branch-BugFixing

Bug fixing

clement559 committed:
Changed method to use toString (newly created to take in predicate and comparator)

clement559 committed:
Renamed countTasksWithFilter to size in TaskList
Changed ProgressBarCommand to use size

erjunze committed:
Merge pull request #118 from clement559/branch-FilterCompletion

List Filters

RuiShengGit committed:
Debug add priority function to recognise missing -edit flag

RuiShengGit committed:
Make changes to if statement for edit priority class

clement559 committed:
Merge pull request #119 from RuiShengGit/branch-EditPriority

Debug edit priority function to recognize missing -edit flag

clement559 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp

clement559 committed:
Added list sorted by priority

clement559 committed:
Missed out files during staging

clement559 committed:
Github check failure

clement559 committed:
Added check if sort filter is not provided (i.e sortMethod == null)

clement559 committed:
Merge pull request #120 from clement559/branch-SortPriority

Branch sort priority

clement559 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp

jerome2406 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp

jerome2406 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp

jerome2406 committed:
Setting the project up for the use of gson library to save data as json

erjunze committed:
Fix more bugs, reformat message constants

erjunze committed:
Swap order of constructor and constants

jerome2406 committed:
Implement storage class using Gson library (with warnings)

erjunze committed:
Merge pull request #122 from ERJUNZE/BugFixes

Fix more bugs, reformat message constants

RuiShengGit committed:
Add help command

jerome2406 committed:
Add TypeAdapter to avoid warnings caused by using Gson library

jeromeongithub committed:
Merge pull request #123 from RuiShengGit/branch-HelpCommand

Add help command

jerome2406 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp into branch-jsonStorage

# Conflicts:
#	src/main/java/seedu/todolist/constants/Messages.java
#	src/main/java/seedu/todolist/exception/FailedLoadException.java
#	src/main/java/seedu/todolist/exception/FailedSaveException.java
#	src/main/java/seedu/todolist/task/TaskList.java
#	src/test/data/properSaveFile.txt

jerome2406 committed:
Delete unnecessary whitespace

jerome2406 committed:
Fix checkstyle issues

clement559 committed:
Merge pull request #124 from jeromeongithub/branch-jsonStorage

Branch json storage

jerome2406 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp

jerome2406 committed:
Modify delete command to have mass delete functionality

jerome2406 committed:
Credit the work of another group which was referenced for json storage

jerome2406 committed:
Add author tag

clement559 committed:
Created configuration file to control the checking and repeating frequency of repeating tasks.
Renamed Kedrian's files to match Command format

clement559 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp into branch-ConfigFile

# Conflicts:
#	src/main/java/seedu/todolist/ToDoListManager.java
#	src/main/java/seedu/todolist/constants/Errors.java
#	src/main/java/seedu/todolist/constants/Flags.java
#	src/main/java/seedu/todolist/constants/Formats.java
#	src/main/java/seedu/todolist/constants/Messages.java
#	src/main/java/seedu/todolist/logic/Parser.java
#	src/main/java/seedu/todolist/storage/Storage.java
#	src/main/java/seedu/todolist/task/TaskList.java
#	src/main/java/seedu/todolist/ui/Ui.java

jerome2406 committed:
Add FailedLoadException and FailedSaveException back after discussion
with teammate

RuiShengGit committed:
Update ppp and make changes to priority levels

jerome2406 committed:
Update Storage Tests

jerome2406 committed:
Change from ArrayList to HashSet to avoid duplicate ids

RuiShengGit committed:
Make changes to class and variable names

jerome2406 committed:
Halfway through converting commands interacting with id to interact
with multiple ids at one go

clement559 committed:
Fix checkstyle errors

jerome2406 committed:
Edit JUnit tests

clement559 committed:
More checkstyle errors

clement559 committed:
Edited text-ui-test EXPECTED to include searching for config file

jerome2406 committed:
Fix bug with editTagsCommand which caused info command to crash after
deleting tags

jerome2406 committed:
Add setPrettyPrinting so that saved file is indented

RuiShengGit committed:
Make changes based on comments

jerome2406 committed:
Change all saved txt files to json files and fix storage tests

erjunze committed:
Merge pull request #125 from RuiShengGit/branch-PPP

Update ppp and make changes to priority levels

jerome2406 committed:
Fix checkstyle issues

jerome2406 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp
into branch-massDelete

# Conflicts:
#	src/main/java/seedu/todolist/ui/Ui.java

jerome2406 committed:
fix checkstyle errors

jerome2406 committed:
fix JUnit tests

jerome2406 committed:
edit gitignore file

clement559 committed:
Edited exceptions accordingly for adding back FailedSaveException and FailedLoadException. Added new FailedSaveConfigException and FailedLoadConfigException.

jerome2406 committed:
fix checkstyle issues

jerome2406 committed:
edit gitignore file to ignore all json files

clement559 committed:
Fix checkstyle errors and remove unused imports

jerome2406 committed:
remove erratic 'z' in file

clement559 committed:
More unused imports

jerome2406 committed:
fix text-ui-test

clement559 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp into branch-ConfigFile

# Conflicts:
#	src/main/java/seedu/todolist/constants/Messages.java
#	src/main/java/seedu/todolist/ui/Ui.java

jerome2406 committed:
Update data.json

jeromeongithub committed:
Merge pull request #128 from clement559/branch-ConfigFile

Branch config file

jerome2406 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp
into branch-massDelete

# Conflicts:
#	src/main/java/seedu/todolist/ToDoListManager.java
#	src/main/java/seedu/todolist/constants/Errors.java
#	src/main/java/seedu/todolist/exception/FailedLoadException.java
#	src/main/java/seedu/todolist/exception/FailedSaveException.java
#	src/main/java/seedu/todolist/exception/InvalidFindException.java
#	src/main/java/seedu/todolist/logic/Parser.java
#	src/main/java/seedu/todolist/storage/Storage.java
#	src/test/java/seedu/todolist/storage/StorageTest.java

jerome2406 committed:
fix checkstyle issues

clement559 committed:
Merge pull request #126 from jeromeongithub/branch-massDelete

Branch mass delete

clement559 committed:
Hotfix for edit config to prevent reset of other variable when editing single variable

clement559 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp into branch-ConfigFile

erjunze committed:
Merge pull request #129 from clement559/branch-ConfigFile

Branch config file

erjunze committed:
Refactor saving data

erjunze committed:
Refactor commands to accomodate config

erjunze committed:
Merge remote-tracking branch 'upstream/master' into idAndFilterSelection

# Conflicts:
#	src/main/java/seedu/todolist/logic/command/EditConfigCommand.java

jerome2406 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp

erjunze committed:
Move frequency parsing to ParserUtil

jerome2406 committed:
Implement selective tag deletion functionality

erjunze committed:
Change priority to use enum

jerome2406 committed:
Change user notification when doing selective tag deletion to be more
informative

erjunze committed:
Move filter and sort parsing to ParserUtil

erjunze committed:
Add more filter options

jerome2406 committed:
Change selective deletion to use switch case instead of if else
statements

jerome2406 committed:
Add default case to switch case statements

jerome2406 committed:
Edit progress command portion of User Guide to clarify what the program
considers as the current week

clement559 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp

jerome2406 committed:
Update portion on storage feature in DG

clement559 committed:
Added personal PPP
Edit UserGuide to include config, list filters and remove proposed completion history feature.

jerome2406 committed:
Update my ppp

jerome2406 committed:
Correctly credit clement's code inside storage class

jerome2406 committed:
Fix typos and indentation in DG

jeromeongithub committed:
Merge pull request #131 from clement559/branch-UG_DG

Addition of personal PPP, small bug fixes

clement559 committed:
Merge pull request #130 from jeromeongithub/branch-selectiveTagDeletion

Branch selective tag deletion

clement559 committed:
Merge pull request #132 from jeromeongithub/branch-PPP

Add personal PPP and @@author tag for clement code in Storage class

clement559 committed:
Merge pull request #133 from jeromeongithub/branch-updateUGandDG

Update UserGuide and DeveloperGuide

clement559 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp

clement559 committed:
Changed filepath for PPP

erjunze committed:
Change multi-ID select to use streams

jeromeongithub committed:
Merge pull request #134 from clement559/branch-UG_DG

Branch ug dg

erjunze committed:
Allow filters in edit, delete, info commands

RuiShengGit committed:
Add improvement to delete command where users can delete all, fix bug of misleading error message and no delete function for priority

RuiShengGit committed:
Fix Checkstyle

RuiShengGit committed:
Fix compilation error for test

RuiShengGit committed:
Fix checkstyle issue

RuiShengGit committed:
Fix expected text

RuiShengGit committed:
Fix expected text

erjunze committed:
Merge remote-tracking branch 'upstream/master' into idAndFilterSelection

Add ability to add tags to existing task

# Conflicts:
#	src/main/java/seedu/todolist/logic/command/EditConfigCommand.java
#	src/main/java/seedu/todolist/logic/command/EditTagsCommand.java
#	src/main/java/seedu/todolist/model/Task.java
#	src/main/java/seedu/todolist/storage/Storage.java
#	src/main/java/seedu/todolist/task/TaskList.java

erjunze committed:
Fix checkstyle issues

erjunze committed:
Fix incorrect access modifiers

erjunze committed:
Add my Project Portfolio Page

RuiShengGit committed:
Merge pull request #137 from ERJUNZE/PPP

Add my Project Portfolio Page

KedrianLoh committed:
Kedrian PPP

KedrianLoh committed:
Merge branch 'master' into Kedrian_PPP

RuiShengGit committed:
Add Kedrian PPP

KedrianLoh committed:
Updates

KedrianLoh committed:
Merge pull request #141 from RuiShengGit/branch-KedrianPPP

Add Kedrian PPP

KedrianLoh committed:
Merge remote-tracking branch 'origin/Kedrian_PPP'

# Conflicts:
#	src/test/data/invalidSaveFile.json

KedrianLoh committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	docs/team/kedrian.md

erjunze committed:
Merge pull request #135 from RuiShengGit/branch-AddDeleteAll

Add improvement to delete command

erjunze committed:
Rename PPP in accordance to requirements

erjunze committed:
Merge pull request #143 from ERJUNZE/PPP

Rename PPP in accordance to requirements

erjunze committed:
Merge branch 'master' into idAndFilterSelection

# Conflicts:
#	src/main/java/seedu/todolist/constants/Errors.java
#	src/main/java/seedu/todolist/constants/Messages.java
#	src/main/java/seedu/todolist/logic/ParserUtil.java
#	src/main/java/seedu/todolist/logic/command/DeleteTaskCommand.java
#	src/main/java/seedu/todolist/logic/command/EditDeadlineCommand.java
#	src/main/java/seedu/todolist/logic/command/EditDescriptionCommand.java
#	src/main/java/seedu/todolist/logic/command/EditEmailCommand.java
#	src/main/java/seedu/todolist/logic/command/EditPriorityCommand.java
#	src/main/java/seedu/todolist/logic/command/EditRepeatCommand.java
#	src/main/java/seedu/todolist/logic/command/EditTagsCommand.java
#	src/main/java/seedu/todolist/logic/command/ListFullInfoCommand.java
#	src/main/java/seedu/todolist/logic/command/MarkTaskCommand.java
#	src/main/java/seedu/todolist/logic/command/UnmarkTaskCommand.java
#	src/main/java/seedu/todolist/task/TaskList.java
#	src/main/java/seedu/todolist/ui/Ui.java
#	text-ui-test/EXPECTED.TXT

erjunze committed:
Move delete all to reset, add more sorts

erjunze committed:
Remove unnecessary catch blocks

erjunze committed:
Fix indentation

erjunze committed:
Fix tests

clement559 committed:
Merge pull request #136 from ERJUNZE/idAndFilterSelection

Add more filters, allow ID-taking commands to also use filters

erjunze committed:
Update help messages

erjunze committed:
Update user guide

erjunze committed:
Remove unused imports

erjunze committed:
Update help section of User Guide

clement559 committed:
Added JavaDoc for most functions

erjunze committed:
Add select by id to list, fix User Guide

clement559 committed:
Merge pull request #146 from ERJUNZE/UserGuide

Update user guide

clement559 committed:
Added neither/both into JavaDoc

clement559 committed:
Fixed all pointed out by Jun Ze

clement559 committed:
Merge branch 'master' into branch-BugFixing

erjunze committed:
Merge pull request #154 from clement559/branch-BugFixing

Added JavaDoc for most functions

erjunze committed:
Add save file validity check

erjunze committed:
Add comments for most classes and public methods

erjunze committed:
Update architecture and component diagrams

erjunze committed:
Update Developer Guide

clement559 committed:
Included mention of configuration settings in check repeating tasks
Removed proposed history feature
Removed new in EditDeadlineCommandSequenceDiagram

erjunze committed:
Merge pull request #157 from clement559/branch-UG_DG

Update Developer Guide & Diagrams

erjunze committed:
Merge remote-tracking branch 'upstream/master' into Comments

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/diagrams/EditDeadlineCommandSequence.puml

erjunze committed:
Update main readme file

erjunze committed:
Fix broken link in DG, remove unused imports

erjunze committed:
Replace placeholder photo

clement559 committed:
Merge pull request #158 from ERJUNZE/Comments

Add comments, update Developer Guide

RuiShengGit committed:
Update Developer Guide and add JUnit tests

RuiShengGit committed:
Fix typo

erjunze committed:
Add authorship tags, update Developer Guide

erjunze committed:
Merge pull request #159 from RuiShengGit/branch-JUnit_Tests

Update Developer Guide and add JUnit tests

jerome2406 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp

erjunze committed:
Merge remote-tracking branch 'upstream/master' into DeveloperGuide

# Conflicts:
#	docs/DeveloperGuide.md

erjunze committed:
Add a test

jerome2406 committed:
Edit personal PPP

jerome2406 committed:
Remove unused method and avoid unnecessary use of this.field

erjunze committed:
Merge pull request #160 from ERJUNZE/DeveloperGuide

Update authorship of developer guide

jerome2406 committed:
Edit documentation for loadData() method

jerome2406 committed:
Add documentation for some methods in Task class

jerome2406 committed:
Fix sequence diagrams by removing incorrect lifeline

jerome2406 committed:
fix checkstyle issue

jerome2406 committed:
Fix indentation error

jerome2406 committed:
fix indentation error

clement559 committed:
Merge pull request #161 from jeromeongithub/branch-Fixes

Fixed some issues mentioned by Wira

erjunze committed:
Update components section of DG

erjunze committed:
Merge remote-tracking branch 'upstream/master' into DeveloperGuide

RuiShengGit committed:
Update PPP

jeromeongithub committed:
Merge pull request #162 from RuiShengGit/branch-PPP

Update PPP

RuiShengGit committed:
Update PPP

jeromeongithub committed:
Merge pull request #163 from RuiShengGit/branch-Update_PPP

Update PPP

erjunze committed:
Update PPP

jeromeongithub committed:
Merge pull request #164 from ERJUNZE/DeveloperGuide

Update Developer Guide

erjunze committed:
Fix links in documentation

clement559 committed:
Merge pull request #165 from ERJUNZE/BugFixes

Fix links in documentation

erjunze committed:
Fix the correct links

clement559 committed:
Merge pull request #166 from ERJUNZE/BugFixes

Fix the correct links in the documentation

erjunze committed:
Fix incorrect file name in instruction

erjunze committed:
Improve instruction

clement559 committed:
Merge pull request #167 from ERJUNZE/BugFixes

Fix incorrect file name in instruction

KedrianLoh committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T11-4/tp

KedrianLoh committed:
Updated PPP

KedrianLoh committed:
Added Junit test

KedrianLoh committed:
Merge branch 'master' into update-PPP

KedrianLoh committed:
Fixed checkstyle errors

KedrianLoh committed:
Merge branch 'update-PPP' of https://github.com/KedrianLoh/tp into update-PPP

KedrianLoh committed:
Fixed PPP error

erjunze committed:
Merge pull request #168 from KedrianLoh/update-PPP

Update ppp

erjunze committed:
Remove unused text file

clement559 committed:
Merge pull request #169 from ERJUNZE/BugFixes

Remove unused text file
